core/OObject.php

```diff
@@ -1031,4 +1003,30 @@ public function getLastException(): Throwable
 	{
 		return $this->lastException;
 	}
+
+	/**
+	 * @param $allowedList
+	 * @param array $list
+	 * @return void
+	 */
+	protected function checkPermissionOrRole($allowedList, array $list): bool
+	{
+		if (!is_array($allowedList)) {
+			return false;
+		} else {
+			foreach ($allowedList as $allowedPermission) {
+				if (!is_string($allowedPermission)) {
+					return false;
+				}
+			}
+
+			$matched = array_intersect($allowedList, $list);
+
+			if (count($matched) === 0) {
+				return false;
+			}
+		}
+
+		return true;
+	}
```

Comment:
Written as a blacklist instead of a whitelist.
```suggestion
	protected function checkPermissionOrRole($allowedList, array $list): bool
	{
		if (is_array($allowedList)) {
			foreach ($allowedList as $allowedPermission) {
				if (!is_string($allowedPermission)) {
					return false;
				}
			}
			if (count(array_intersect($allowedList, $list)) > 0) {
				return true;
			}
		}
		return false;
	}
```
Or if we don't care about the array_intersect optimization (I think I prefer this option)
```suggestion
	protected function checkPermissionOrRole1($allowedList, array $list): bool
	{
		if (is_array($allowedList)) {
			foreach ($allowedList as $allowedPermission) {
				if (in_array($allowedPermission, $list, true)) {
					return false;
				}
			}
		}
		return false;
	}
```